src/SortChain.php

```diff
@@ -42,6 +48,10 @@ public function keys($collection)
 
     public function __invoke($a, $b)
     {
+        if (empty($this->callables)) {
+            return static::$nullOperatorEquality;
```

Comment:
True, though in that case equality will typically be done on the basis of a compare() or equality() method on the object in question. Otherwise, if they're falling back on a loose type check, I'd argue it's performing correctly.

Either way, I've added an extra test for the "mostly homogeneous scenario" but it seems stable enough to merge for now. Thanks for the feedback!